doc/getting-started.rst

```diff
@@ -198,6 +209,18 @@ the following file named ``myscript``:
     main :: IO ()
     main = haskellSay "Hello, Haskell!"
 
+.. note::
+
+    Widening or dropping version bound constraints on *packages included with
+    the compiler* [#boot-packages]_, like ``base``, will allow single-file
+    scripts to run with a wider range of compiler versions.
```

Comment:
It is still too optimistic w.r.t. running with other versions of compiler because the script can be using something very specific to the current compiler version (e..g a particular language extension in the latest version of the compiler).
```suggestion
    the compiler* [#boot-packages]_, like ``base``, may allow single-file
    scripts to run with a wider range of compiler versions.
```